skimage/segmentation/tests/test_slic.py

```diff
@@ -208,5 +212,254 @@ def test_more_segments_than_pixels():
+    assert_equal(seg[:, :2], 0)
+    assert_equal(seg[:, -2:], 0)
+
+
+def test_list_sigma_mask():
+    rnd = np.random.RandomState(0)
+    msk = np.zeros((2, 6))
+    msk[:, 1:-1] = 1
+    img = np.array([[1, 1, 1, 0, 0, 0],
+                    [0, 0, 0, 1, 1, 1]], np.float)
+    img += 0.1 * rnd.normal(size=img.shape)
+    result_sigma = np.array([[-1, 0, 0, 1, 1, -1],
+                             [-1, 0, 0, 1, 1, -1]], np.int) + 1
+    seg_sigma = slic(img, n_segments=2, sigma=[1, 50, 1],
+                     multichannel=False, mask=msk)
+    assert_equal(seg_sigma, result_sigma)
+
+
+def test_spacing_mask():
+    rnd = np.random.RandomState(0)
+    msk = np.zeros((2, 5))
+    msk[:, 1:-1] = 1
+    img = np.array([[1, 1, 1, 0, 0],
+                    [1, 1, 0, 0, 0]], np.float)
+    result_non_spaced = np.array([[-1, 0, 0, 1, -1],
+                                  [-1, 0, 1, 1, -1]], np.int) + 1
+    result_spaced = np.array([[-1, 0, 0, 0, -1],
+                              [-1, 1, 1, 1, -1]], np.int) + 1
+    img += 0.1 * rnd.normal(size=img.shape)
+    seg_non_spaced = slic(img, n_segments=2, sigma=0, multichannel=False,
+                          compactness=1.0, mask=msk)
+    seg_spaced = slic(img, n_segments=2, sigma=0, spacing=[1, 500, 1],
+                      compactness=1.0, multichannel=False, mask=msk)
+    assert_equal(seg_non_spaced, result_non_spaced)
+    assert_equal(seg_spaced, result_spaced)
+
+
+def test_enforce_connectivity_mask():
+    msk = np.zeros((3, 6))
+    msk[:, 1:-1] = 1
+    img = np.array([[0, 0, 0, 1, 1, 1],
+                    [1, 0, 0, 1, 1, 0],
+                    [0, 0, 0, 1, 1, 0]], np.float)
+
+    segments_connected = slic(img, 2, compactness=0.0001,
+                              enforce_connectivity=True,
+                              convert2lab=False, mask=msk)
+    segments_disconnected = slic(img, 2, compactness=0.0001,
+                                 enforce_connectivity=False,
+                                 convert2lab=False, mask=msk)
+
+    # Make sure nothing fatal occurs (e.g. buffer overflow) at low values of
+    # max_size_factor
+    segments_connected_low_max = slic(img, 2, compactness=0.0001,
+                                      enforce_connectivity=True,
+                                      convert2lab=False,
+                                      max_size_factor=0.8, mask=msk)
+
+    result_connected = np.array([[-1, 0, 0, 1, 1, -1],
+                                 [-1, 0, 0, 1, 1, -1],
+                                 [-1, 0, 0, 1, 1, -1]], np.float) + 1
+
+    result_disconnected = np.array([[-1, 0, 0, 1, 1, -1],
+                                    [-1, 0, 0, 1, 1, -1],
+                                    [-1, 0, 0, 1, 1, -1]], np.float) + 1
+
+    assert_equal(segments_connected, result_connected)
+    assert_equal(segments_disconnected, result_disconnected)
+    assert_equal(segments_connected_low_max, result_connected)
+
+
+def test_slic_zero_mask():
```

Comment:
Do you have a test that compares slic to maskSLIC for a pass-through mask?

Reply:
The two implementations uses different approaches for the centroid initialization, so I guess that the output will be different...